### PR TITLE
add NegotiatedCipherSuite to QuicConnection

### PR DESCRIPTION
**Pull request by wfurt**
contributes to #70184

### CODE REVIEW

**Comment by wfurt** [comment]
The server side is failing for no good reason. I opened https://github.com/microsoft/msquic/issues/4457 for tracking.

File: src/libraries/System.Net.Quic/tests/FunctionalTests/QuicConnectionTests.cs [hunk 1]
```diff
@@ -47,10 +48,13 @@ public async Task TestConnect(IPAddress address)
                 // https://github.com/microsoft/msquic/issues/3813
                 Assert.Equal(clientConnection.LocalEndPoint, serverConnection.RemoteEndPoint);
             }
-            Assert.Equal(ApplicationProtocol.ToString(), clientConnection.NegotiatedApplicationProtocol.ToString());
-            Assert.Equal(ApplicationProtocol.ToString(), serverConnection.NegotiatedApplicationProtocol.ToString());
```
Comment (rzikm) [inline src/libraries/System.Net.Quic/tests/FunctionalTests/QuicConnectionTests.cs:51]: Is there a reason for removing these?

**Review by rzikm — COMMENTED** [review]
The API proposal also added SslProtocol property, Will you add it as well in this PR?

**Comment by rzikm** [comment]
I think we need to call the getter proactively during QUIC_CONNECTION_EVENT_COMPLETED as documentation suggests

https://github.com/microsoft/msquic/blob/f96015560399d60cbdd8608b6fa2120560118500/docs/Settings.md#L186-L193

MsQuic drops TLS state after the handshake is complete.

**Pull request closed without merging**